dynamiqs/mesolve/mesolve.py

```diff
@@ -129,22 +124,22 @@ def _vmap_mesolve(
 ) -> MEResult:
     # === vectorize function
     # we vectorize over H, jump_ops and rho0, all other arguments are not vectorized
-    is_batched = (
-        is_timearray_batched(H),
-        [is_timearray_batched(jump_op) for jump_op in jump_ops],
-        rho0.ndim > 2,
-        False,
-        False,
-        False,
-        False,
-        False,
+    n_batch = (
+        H.ndim - 2,
+        [jump_op.ndim - 2 for jump_op in jump_ops],
+        rho0.ndim - 2,
+        0,
+        0,
+        0,
+        0,
+        0,
     )
 
     # the result is vectorized over `_saved` and `infos`
     out_axes = MEResult(None, None, None, None, 0, 0)
 
     # compute vectorized function with given batching strategy
-    f = compute_vmap(_mesolve, options.cartesian_batching, is_batched, out_axes)
+    f = compute_vmap(_mesolve, options.cartesian_batching, n_batch, out_axes)
```

Comment:
I guess we're missing a call to `as_batched_callable` for `H` and `jump_ops` somewhere before this? This is what's done for `sesolve`.